### DISCUSSION TITLE
Compiling debug logs, but set log level to info by default

### DISCUSSION THREAD

**Discussion by toonst** [zephyrproject-rtos/zephyr#57957] [Q&A] [answered]
For a specific module, I would like to have debug logging compiled in, so I can enable them through the shell with
`log enable dbg my_module'.
However, I don't want to have the debug logs when the device boots.
Is there a way to configure compiled in log levels differently than default log levels per module?

**Comment by toonst** [comment, marked as answer]
It seems there is no way to do this through Kconfig alone. The solution was to set the log levels at runtime from application code.
```
#include <zephyr/init.h>
#include <zephyr/logging/log_ctrl.h>

typedef struct {
    const char* module;
    uint32_t level;
} log_module_t;

static log_module_t log_modules[] = {
    {"net_if", LOG_LEVEL_INF},
    {"net_ethernet", LOG_LEVEL_INF},
    // add more modules as needed
};

static int set_initial_log_levels(const struct device *dev)
{
    ARG_UNUSED(dev);

    for (unsigned int i = 0; i < ARRAY_SIZE(log_modules); i++) {
        int source_id = log_source_id_get(log_modules[i].module);
        if (source_id < 0) {
            printk("Error: Log module %s not found\n", log_modules[i].module);
            continue;
        }
        log_filter_set(NULL, 0, source_id, log_modules[i].level);
    }

    return 0;
}

SYS_INIT(set_initial_log_levels, APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY);

```

**Reply by aburo8** [reply]
I understand that this should work in theory, but it is not working for me. The DBG logs are still being printed even after setting the log level to INF as you have done in this code snippet.  Are there any reasons you can think of which could cause this to happen? What are your prj.conf settings?
See #55703 - I believe this is the issue I am encountering since I am using the shell module.
